### DISCUSSION TITLE
How to use CapacitorBridge Class

### DISCUSSION THREAD

**Discussion by skqew** [ionic-team/capacitor#4806] [Q&A]
I want to call CapacitorBridge
But CapacitorBridge is internal class
How do I call the CapacitorBridge class and use getCommandInstance() ?

func getCommandInstance(pluginName: String) -> CDVPlugin? {

    if let viewController = UIApplication.shared.keyWindow?.rootViewController as? CAPBridgeViewController {
        
        return (viewController.bridge?.cordovaPluginManager?.getCommandInstance(pluginName))!
        -> error : Value of type 'CAPBridgeProtocol' has no member 'cordovaPluginManager'
    }

    return nil
}

**Comment by jcesarmobile** [comment]
Why do you want o to use it?
getCommandInstance is a method of the cordova compatibility classes, used internally by capacitor to make cordova plugins work.

**Reply by conkyliu** [reply]
> 为什么要o使用它？ getCommandInstance 是cordova 兼容类的一种方法，由电容器在内部使用以使cordova 插件工作。

Because the call I use to handleOpen fails and must pass getCommnadInstance

`
if url.scheme == "xxxx" {
            
            let wechat = getCommnadInstance(pluginName: "wechat") as! CDVWechat;
            return wechat.handleOpen(url)
            
        }
`
`func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:])`
call this function

**Reply by rlindgren** [reply]
@jcesarmobile What if we need to access `cordovaPluginManager.getCommandInstance`?  is there any way to do so?
